File: Java_Codes/15. Memory Management/MemoryLeakExample.java
//Commonly observed in applications that store large collections 
// (e.g., caching data) without proper cleanup mechanisms.

import java.util.ArrayList;
import java.util.List;

public class MemoryLeakExample {

    private static List<Object> memoryLeak = new ArrayList<>();

    public static void demonstrateMemoryLeak() {
        for (int i = 0; i < 100000; i++) {
            memoryLeak.add(new Object()); // Objects are never removed
        }
        System.out.println("Objects added without release. Potential memory leak!");
    }

    public static void main(String[] args) {
        demonstrateMemoryLeak();
    }
}

File: Projects/PythonProject/Memory_Management/MemoryLeakExample.py
# Simulating a memory leak by holding references in a global list

memory_leak = []

def demonstrate_memory_leak():
    for i in range(100000):
        memory_leak.append(object())  # Objects are never released
    print("Objects added without release. Potential memory leak!")

if __name__ == "__main__":
    demonstrate_memory_leak()
